File: 10001st prime/10001st_prime.py
# function to verify if number is prime
def primeNumber(number):
    numberHalf = number/2
    divisor = 1
    divisorList = []

    while divisor <= numberHalf:
        if number % divisor == 0:
            divisorList.append(divisor)
        divisor = divisor + 1

    if len(divisorList) == 1:
        return True
    else:
        return False

def nth_prime(nth):
    prime = 1
    primeList = []

    while len(primeList) < nth:
        if primeNumber(prime) == True:
            primeList.append(prime)
        prime = prime + 1
    print(f'Size {len(primeList)}.')

    return max(primeList)

nth = int(input("How many elements? "))

# the challenge asks for 10001st element
print(f'The {nth} prime is {nth_prime(nth)}.')

File: 10001st prime/10001st_prime.cpp
#include <iostream>
#include <vector>
#include <algorithm>

// function to verify if number is prime
bool primeNumber(int number)
{
    int numberHalf = number / 2;
    int divisor = 1;
    std::vector<int> divisorList;

    while (divisor <= numberHalf)
    {
        if (number % divisor == 0)
        {
            divisorList.push_back(divisor);
        }
        divisor = divisor + 1;
    }

    if (divisorList.size() == 1)
    {
        return true;
    }
    else
    {
        return false;
    }
}

int nth_prime(int nth)
{
    int prime = 1;
    std::vector<int> primeList;

    while (primeList.size() < nth)
    {
        if (primeNumber(prime) == true)
        {
            primeList.push_back(prime);
        }
        prime = prime + 1;
    }
    std::cout << "Size " << primeList.size() << ".\n";

    int max = *max_element(primeList.begin(), primeList.end());

    return max;
}

/*int main()
{
    int nth;
    std::cout << "How many elements? ";
    std::cin >> nth;

    // the challenge asks for 10001st element
    std::cout << nth_prime(nth);

    return 0;
}*/